Fix: #970. Digest semantic depends on method. (#1248)
httpwg/http-extensions: draft-ietf-httpbis-digest-headers.md

   example an incremental digest-algorithm. A document defining such a
   digest-algorithm is best positioned to describe how it is used.

9. Why don't we say that in requests, Digest is always computed on the payload body?

>    This [discussion is not trivial](https://github.com/httpwg/http-extensions/pull/923)
>    because while in both POST and PATCH the representation-digest is computed on the payload body,
>    this is because of the semantics of those methods. For example, an hypothetical method where
>    the request body contains a partial representation would behave differently, and Digest should
>    be computed on the complete representation data. An example of such an hypothetical use case
>    would be a resumable upload mechanism implemented with PUT + Content-Range like
>    [this one](https://cloud.google.com/storage/docs/performing-resumable-uploads#resume-upload).

Reviewer: ```suggestion
   This [discussion is not trivial](https://github.com/httpwg/http-extensions/pull/923).
   When using POST and PATCH the representation-digest is computed on the payload body but
   that this is because of the semantics of those methods. However, other method semantics can vary.
   For example, a hypothetical method where the request body contains a partial representation
   would behave differently, and it's Digest would be computed on the complete representation data.
   An example of such an hypothetical use case would be a resumable upload mechanism 
   implemented with PUT + Content-Range, such as
   [this one](https://cloud.google.com/storage/docs/performing-resumable-uploads#resume-upload).
```

just some editorializing

Reviewer 2: No. 

The Digest shoult not depend on the method. And giving an example of something that the HTTP spec explicitly forbids (PUT + Range) doesn't make it better.

Author: > No.

Can you provide some normative references in SEMANTICS which forbids to create a method or a mechanism conveying a partial representation? It would be really useful :) 

> [..] HTTP spec explicitly forbids (PUT + Range)

let it be an `UPDATE` method which conveys a partial representation instead of a patch-document. 
Probably the proliferation of different ways of updating resources via partial representations and the (mis)use of Content-MD5 by goog, amz & co requires some attention from us.

> Digest should not depend on the method

Digest does not depend directly on the method, but on the SEMANTICS (which is impacted by the method).
imho stating that Digest just depends on the payload body is not in line with RFC3230 spirit.
